### PR TITLE
Add desired mac settings to end device general settings in the Console

### LINKED ISSUE #4290: Change some default values in the Console
#### Summary

Change some default values in the Console

#### Why do we need this?

To make sure that we have saner defaults when people don't change the values.

#### What is already there? What do you see now?

1. RX1 delay defaults to 1 second when not entered. This happens at least when creating ABP devices
2. Gateway status is not public by default. There is no explanation what this means
3. Gateway location is not public by default. There is no explanation what this means

#### What is missing? What do you want to see?

1. Default RX1 delay should be 5 seconds, and this should also be displayed in the input box
2. Gateway status should be public by default. There should also be a short description that only if the location is public, the gateway status is shown on maps. Mapping services (Packet Broker, TTN Mapper) may have their own access policies (not sure if we need to mention the mapping services by name or just explain the concept)
3. Gateway location should be public by default. See (2) for explanation

#### How do you propose to implement this?

This should probably be baked in the Console, but I can also imagine that we're missing a default value in the NS (?). Also check how CLI handles this

#### How do you propose to test this?

Local

#### Can you do this yourself and submit a Pull Request?

No

### PR DESCRIPTION
**Pull request by bafonins**
#### Summary

Closes https://github.com/TheThingsNetwork/lorawan-stack/issues/4290 and https://github.com/TheThingsNetwork/lorawan-stack/issues/2047

<details>
<summary>Screenshots</summary>

# OTAA:

## default

otaa-ns-settings

## class B

otaa-class-b-ns-settings

## class C

otaa-class-c-ns-settings

# ABP:

## default

abp-ns-settings

## default lorawan version 1.1.0

abp-lw-110-ns-settings

## class B

abp-class-b-ns-settings

## class C

abp-class-c-ns-settings

# MULTICAST:

## class B

multicast-class-b-ns-settings

## class C

multicast-class-c-ns-settings

</details>

#### Changes

- Add desired mac settings to end device general settings
- Update field titles
- Add tooltips for missing fields
- Remove descriptions for fields with tooltip content
- Add starting point for e2e tests

#### Testing

Manual

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/webui/console/views/device-general-settings/network-server-form/index.js
Comment (kschiffer) [inline pkg/webui/console/views/device-general-settings/network-server-form/index.js]:
```suggestion
  const handlePhyVersionChange = React.useCallback(setPhyVersion, [])
```

**Comment by kschiffer** [comment]
I believe this needs attention from @johanstokking, to make sure the fields added here are correct and logical (specifically having both desired and actual values). So I'd suggest putting this on hold until he's back. Code-wise, it looks good to me.

**Comment by johanstokking** [comment]
This looks good to me for ABP.

However, for OTAA, I miss most of the settings. I clarified https://github.com/TheThingsNetwork/lorawan-stack/issues/4290#issuecomment-888136034; for OTAA we need to configure a bunch of desired parameters as well.

**Comment by bafonins** [comment]
@johanstokking So how should this look for OTAA?
Why should we allow editing only the `_desired` fields for OTAA devices and not the actual parameters, e.g. `mac_settings.rx2_data_rate_index` and `mac_settings.desired_rx2_data_rate_index`?

**Comment by johanstokking** [comment]
> Why should we allow editing only the `_desired` fields for OTAA devices and not the actual parameters, e.g. `mac_settings.rx2_data_rate_index` and `mac_settings.desired_rx2_data_rate_index`?

Because RX2 data rate is part of the join-accept in OTAA. So it is already negotiated between NS and the end-device. Allowing users to customize it in the NS breaks downlink. The current value should not be changed.

With ABP there's no negotiation on this, so it should be configurable.

The desired value can always be configured; through MAC commands over downlink, the RX data rate index can be changed.

**Comment by bafonins** [comment]
@johanstokking  just updated screenshots. Can you take a look and check if all the fields are present for each activation mode?

**Comment by johanstokking** [comment]
Looks great.

@adriansmares can you check this as well please?

**Review by adriansmares — APPROVED** [review]
Looks good to me too. 

Should we mark every `current_parameters` as being current (`RX1 delay` -> `Current RX1 delay`) ?  I realize this is scope creep so feel free to do it in another PR.

**Comment by johanstokking** [comment]
I'm following @adriansmares' judgement here and it looks like he already approved.

**Review by adriansmares — APPROVED** [review]

**Review by kschiffer — APPROVED** [review]

File: pkg/webui/console/components/mac-settings-section/index.js
Comment (kschiffer) [inline pkg/webui/console/components/mac-settings-section/index.js:213]:
```suggestion
              tooltipId={tooltipIds.RX1_DELAY}
              name="mac_settings.desired_rx1_delay"
```

We can use the same tooltip.

File: pkg/webui/console/components/mac-settings-section/index.js
Comment (kschiffer) [inline pkg/webui/console/components/mac-settings-section/index.js:239]:
```suggestion
              component={Input}
              tooltipId={tooltipIds.DATA_RATE_OFFSET}
```

File: pkg/webui/console/components/mac-settings-section/index.js
Comment (kschiffer) [inline pkg/webui/console/components/mac-settings-section/index.js:274]:
```suggestion
          tooltipId={tooltipIds.RX2_DATA_RATE_INDEX}
          component={Input}
```

File: pkg/webui/console/components/mac-settings-section/index.js
Comment (kschiffer) [inline pkg/webui/console/components/mac-settings-section/index.js:301]:
```suggestion
          tooltipId={tooltipIds.RX2_FREQUENCY}
          component={Input}
```

File: pkg/webui/console/components/mac-settings-section/index.js
Comment (kschiffer) [inline pkg/webui/console/components/mac-settings-section/index.js:309]: This one is missing a tooltip. Can we add one? You can use the same for `desired_max_duty_cycle` as well then.

File: pkg/webui/console/components/mac-settings-section/index.js
Comment (kschiffer) [inline pkg/webui/console/components/mac-settings-section/index.js:411]:
```suggestion
              name="mac_settings.desired_ping_slot_frequency"
              tooltipId={tooltipIds.PING_SLOT_FREQUENCY}
```

File: pkg/webui/console/components/mac-settings-section/index.js
Comment (kschiffer) [inline pkg/webui/console/components/mac-settings-section/index.js:432]:
```suggestion
              tooltipId={tooltipIds.PING_SLOT_DATA_RATE_INDEX}
              component={Input}
```

Comment (kschiffer) [inline pkg/webui/console/components/mac-settings-section/index.js:432]:
```suggestion
              tooltipId={tooltipIds.PING_SLOT_DATA_RATE_INDEX}
              component={Input}
```

File: pkg/webui/console/components/mac-settings-section/index.js
Comment (kschiffer) [inline pkg/webui/console/components/mac-settings-section/index.js]: Can we also add a tooltip for this one?
cc @johanstokking can you provide?

File: pkg/webui/console/components/mac-settings-section/index.js
Comment (kschiffer) [inline pkg/webui/console/components/mac-settings-section/index.js]: Same here.

File: pkg/webui/console/components/mac-settings-section/index.js
Comment (kschiffer) [inline pkg/webui/console/components/mac-settings-section/index.js]: Please add a tooltip for this:

> Controls whether the device uses adaptive data rate. This will allow the network to adjust the employed data rate based on signal to noise ratio. This adaptively optimizes energy consumption, bandwith and transmission power.

Add a documentation link to https://www.thethingsindustries.com/docs/reference/adr

**Review by kschiffer — COMMENTED** [review]
Please see my comment on adding more tooltips for the fields. If possible please add those before merging. There are a couple `desired_*` fields that can use the same tooltip as the normal field.  
Please source the missing tooltip texts from @johanstokking's comment [here](https://github.com/TheThingsNetwork/lorawan-stack/issues/4290#issuecomment-887790141).

File: pkg/webui/console/components/mac-settings-section/index.js
Comment (kschiffer) [inline pkg/webui/console/components/mac-settings-section/index.js]: Please add a tooltip for this:

> Signal-to-noise ratio (SNR) margin in dB that is taken into account in the Adaptive Data Rate (ADR) algorithm to optimize the data rate of the end device. A higher margin requires the end device to have a better SNR before the Network Server instructs a higher data rate for the end device to use.

(via @johanstokking )

Add a documentation link to [https://www.thethingsindustries.com/docs/reference/adr](https://www.thethingsindustries.com/docs/reference/adr)

**Pull request merged**
